### PR TITLE
special case responseType java.lang.String to be a ds3_str

### PR DESCRIPTION
**Pull request by DenverM80**
add generation of static files to CCodeGenerator

tweaks for new functions and types in templates

change SdkName for BucketApiBean from Ds3Bucket to BucketDetails

change SdkName for NodeApiBean from Ds3Node to Endpoint

update all tests to reflect the previous 2 changes

### CODE REVIEW

**Comment by DenverM80** [comment]
2380 build errors reduce down to 1803 with these changes

File: ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/converters/RequestConverter.java [hunk 1]
```diff
@@ -178,19 +178,26 @@ private static boolean isResourceIdRequired(final Ds3Request ds3Request) {
...
         for (final Ds3ResponseCode responseCode : responseCodes) {
             final int rc = responseCode.getCode();
             if (rc < 200 || rc >= 300) continue;
 
             for (final Ds3ResponseType responseType : responseCode.getDs3ResponseTypes()) {
                 if (ConverterUtil.hasContent(responseType.getType()) && !responseType.getType().contentEquals("null")) {
-                    return StructHelper.getResponseTypeName(responseType.getType());
+                    responseTypeName = responseType.getType();
                 }
                 if (ConverterUtil.hasContent(responseType.getComponentType()) && !responseType.getComponentType().contentEquals("null")) {
```
Comment (RachelTucker) [inline ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/converters/RequestConverter.java]: Should this be an else if where you first check if the response type has a component type, else check the type?

File: ds3-autogen-parser/src/main/resources/typeNameMap.json [hunk 2]
```diff
@@ -74,7 +74,7 @@
     },
     {
       "ContractName": "NodeApiBean",
-      "SdkName": "Ds3Node"
+      "SdkName": "Endpoint"
```
Comment (RachelTucker) [inline ds3-autogen-parser/src/main/resources/typeNameMap.json]: I'm not sure if renaming the Ds3Node -> Endpoint is clear. Maybe rename to NodeDetails instead...?

**Comment by rpmoore** [comment]
You're build failed

**Comment by DenverM80** [comment]
I thought all tests were passing but I was in a bit of a hurry to get out the door, looking now...

File: ds3-autogen-parser/src/main/resources/typeNameMap.json (hunk 2 above)
Comment (DenverM80) [inline ds3-autogen-parser/src/main/resources/typeNameMap.json]: they are sort of overlapping, not quite a superset in the way that bucket vs bucketDetails was... but I'm def open to changing it.. I'll ponder this a bit more

File: ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/converters/RequestConverter.java (hunk 1 above)
Comment (DenverM80) [inline ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/converters/RequestConverter.java]: yeah, that would be clearer.  will fix

**Pull request merged**